File: server/FitMate.Core/JsonModels/Exercises/ExerciseModel.cs
namespace FitMate.Core.JsonModels.Exercises;

public class ExerciseModel
{
    public long Id { get; set; }
    public long? UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public string? VideoUrl { get; set; }
    public long PrimaryMuscleGroupId { get; set; }
    public long? SecondaryMuscleGroupId { get; set; }
    public DateTime DateCreated { get; set; }
    public DateTime? DateModified { get; set; }
}

File: client/src/types/JsonModels/Exercises/ExerciseModel.ts
import type { ExerciseCategory } from "../Enums/ExerciseCategory";
import type { ExerciseDifficulty } from "../Enums/ExerciseDifficulty";
import type { ExerciseEquipment } from "../Enums/ExerciseEquipment";
import type { ExerciseLoadBasis } from "../Enums/ExerciseLoadBasis";
import type { ExerciseMovementPattern } from "../Enums/ExerciseMovementPattern";

export interface ExerciseModel
{
	id: number;
	userId?: number;
	isPublic: boolean;
	name: string;
	slug: string;
	description?: string;
	imageUrl?: string;
	videoUrl?: string;
	primaryMuscleGroupId: number;
	secondaryMuscleGroupId?: number;
	equipment?: ExerciseEquipment;
	movementPattern?: ExerciseMovementPattern;
	difficulty?: ExerciseDifficulty;
	category?: ExerciseCategory;
	loadBasis?: ExerciseLoadBasis;
	aliases: string[];
	creatorDisplayName?: string;
	dateCreated: string;
	dateModified?: string;
}
